presto-spi/src/main/java/com/facebook/presto/spi/type/TimeZoneKey.java

```diff
@@ -137,7 +139,10 @@ public static TimeZoneKey getTimeZoneKeyForOffset(long offsetMinutes)
             return UTC_KEY;
         }
 
-        checkArgument(offsetMinutes >= OFFSET_TIME_ZONE_MIN && offsetMinutes <= OFFSET_TIME_ZONE_MAX, "Invalid offset minutes %s", offsetMinutes);
+        if (offsetMinutes < OFFSET_TIME_ZONE_MIN || offsetMinutes > OFFSET_TIME_ZONE_MAX) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid offset minutes " + offsetMinutes);
```

Comment:
Not sure this exception should be raised here. Note that for example 5 lines below we throw `TimeZoneNotSupportedException` and not a `PrestoException`.
How do you know `offsetMinutes` is exactly some Presto function's argument and does not come from some intermediate calculation?

Reply:
First question: where to throw the exception

There are a few places that calls packDateTimeWithZone(long millisUtc, int offsetMinutes) and getTimeZoneKeyForOffset(offsetMinutes):

1. fromUnixTime() : this is the Presto function
2. DateTimeFunctions.timeAtTimeZone( long timeWithTimeZone, long zoneOffset) : this function is not referenced by anybody else
3. DateTimeFunctions.timestampAtTimeZone(long timestampWithTimeZone, long zoneOffset) : this function is not referenced by anybody else either
4. DateTimeJsonValueProvider.getLong() : this only calls packDateTimeWithZone(millis, 0);
5. ISO8601HashRedisValueProvider.getLong() : this only calls packDateTimeWithZone(millis, 0);

So the only meaningful reference is fromUnixTime(). 

We have an alternative to move the throwing of the exception to packDateTimeWithZone(long millisUtc, int offsetMinutes), but this function is just calling 
return packDateTimeWithZone(millisUtc, getTimeZoneKeyForOffset(offsetMinutes));
So it doesn't make much difference than calling from inside getTimeZoneKeyForOffset().

@nezihyigitbasi what's your opinion on this?

Second question. Whether we should throw an PrestoException or a separate exception like TimeZoneNotSupportedException: 
We have seen the reference comes from fromUnixTime(), and we have already defined the error code for it, so we tend to use the existing PrestoException with INVALID_FUNCTION_ARGUMENT.